dot/network/utils.go

```diff
@@ -204,6 +204,7 @@ func readStream(stream libp2pnetwork.Stream, buf []byte) (int, error) {
 		return 0, fmt.Errorf("message size greater than maximum: got %d", length)
 	}
 
+	// TODO: Allocate buffer after reading length. Avoid pre allocation.
```

Comment:
can you remove this? the reason it pre-allocates is because:
1. allocation is expensive, if we allocate each time we receive a message it increases our CPU usage a lot
2. we don't want to allocate a byte array based on a length a peer gives us, because a malicious peer can then tell us to allocate some very large buffer causing the node to crash